python/packages/aisearch/agent_framework_aisearch/_search_provider.py

```diff
@@ -0,0 +1,850 @@
+        embedding_function: Callable[[str], Awaitable[list[float]]] | None = None,
+        context_prompt: str | None = None,
+        # Agentic mode parameters (Knowledge Base)
+        azure_ai_project_endpoint: str | None = None,
+        azure_openai_resource_url: str | None = None,
+        model_deployment_name: str | None = None,
+        model_name: str | None = None,
+        knowledge_base_name: str | None = None,
+        retrieval_instructions: str | None = None,
+        azure_openai_api_key: str | None = None,
+        knowledge_base_output_mode: Literal["extractive_data", "answer_synthesis"] = "extractive_data",
+        retrieval_reasoning_effort: Literal["minimal", "medium", "low"] = "minimal",
+        agentic_message_history_count: int = _DEFAULT_AGENTIC_MESSAGE_HISTORY_COUNT,
+    ) -> None:
+        """Initialize Azure AI Search Context Provider.
+
+        Args:
+            endpoint: Azure AI Search endpoint URL.
+            index_name: Name of the search index to query.
+            credential: Azure credential (API key string, AzureKeyCredential, or AsyncTokenCredential).
+                If a string is provided, it will be converted to AzureKeyCredential.
+            settings: Settings object with auto-loaded configuration from environment.
+                If provided, endpoint/index_name/credential from settings will be used as defaults.
+            mode: Search mode - "semantic" for hybrid search with semantic ranking (fast)
+                or "agentic" for multi-hop reasoning (slower). Default: "semantic".
+            top_k: Maximum number of documents to retrieve. Only applies to semantic mode.
+                In agentic mode, the server-side Knowledge Base determines retrieval based on
+                query complexity and reasoning effort. Default: 5.
+            semantic_configuration_name: Name of semantic configuration in the index.
+                Required for semantic ranking. If None, uses index default.
+            vector_field_name: Name of the vector field in the index for hybrid search.
+                Required if using vector search. Default: None (keyword search only).
+            embedding_function: Async function to generate embeddings for vector search.
+                Signature: async def embed(text: str) -> list[float]
+                Required if vector_field_name is specified and no server-side vectorization.
+            context_prompt: Custom prompt to prepend to retrieved context.
+                Default: "Use the following context to answer the question:"
+            azure_ai_project_endpoint: Azure AI Foundry project endpoint URL.
+                Required for agentic mode. Example: "https://myproject.services.ai.azure.com"
+            model_deployment_name: Model deployment name in the Azure AI project.
+                Required for agentic mode.
+            model_name: The underlying model name (e.g., "gpt-4o", "gpt-4o-mini").
```

Comment:
why do we need both? I'm sure there is a good reason, so let's make sure we clarify that for the user